### PR TITLE
Add dedicated USE_BOOST_DEFAULT_OPTIONS

### PR DESCRIPTION
**Pull request by jcarpent**
Compared to #440, this PR introduces a dedicated macro USE_BOOST_DEFAULT_OPTIONS with some minor fix. This will help for the transition.

### CODE REVIEW

File: boost.cmake [hunk 1]
```diff
@@ -36,6 +36,28 @@ IF(CMAKE_VERSION VERSION_LESS "3.12")
...
+    FIND_PACKAGE(Boost ${BOOST_REQUIRED})
+  ENDIF(NOT Boost_FOUND)
+  STRING(REPLACE "_" "." Boost_SHORT_VERSION ${Boost_LIB_VERSION})
+  SET(Boost_USE_STATIC_LIBS OFF)
+  SET(Boost_USE_MULTITHREADED ON)
+  IF("${Boost_SHORT_VERSION}" VERSION_GREATER "1.70" OR "${Boost_SHORT_VERSION}" VERSION_EQUAL "1.70")
+    SET(BUILD_SHARED_LIBS ON) 
+    SET(Boost_NO_BOOST_CMAKE ON) 
+  ENDIF("${Boost_SHORT_VERSION}" VERSION_GREATER "1.70" OR "${Boost_SHORT_VERSION}" VERSION_EQUAL "1.70")
+ENDMACRO(USE_BOOST_DEFAULT_OPTIONS)
```
Comment (jmirabel) [inline boost.cmake:59]: The name of the function is not clear. If the function only setup the default options, then there shouldn't be a `find_package` inside.

Is this an internal function or is it meant to be used outside ?

If it is meant to be used outside, I would prefer to move the find_package outside, if possible. If not, then you should rename the macro.

Comment (jcarpent) [reply boost.cmake:48]: I removed any dependency to find_package. I did think it would work but it seems to work well.

Comment (jmirabel) [reply boost.cmake:48]: Thanks.

Comment (jmirabel) [reply boost.cmake]: In the past, macro `search_for_boost` added lines to the configured Config.cmake. Is this still needed ?

Comment (jcarpent) [reply boost.cmake]: Yes, you're right. This even mandatory.

Comment (jcarpent) [reply boost.cmake]: @jmirabel Maybe, we should add it to `ADD_PROJECT_DEPENDENCY` which will detect the special case of Boost. Or do you think we add it if we include boost.cmake anyway?

Comment (jcarpent) [reply boost.cmake]: I've added a new function, which should be called when required.

Comment (jmirabel) [reply boost.cmake]: I prefer to let users explicitly write it in their CMakeLists so a macro, as you did, is better in my opinion.

**Pull request merged**